sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -465,6 +465,8 @@ case class DataSource(
     providingClass.newInstance() match {
       case dataSource: CreatableRelationProvider =>
         SaveIntoDataSourceCommand(data, dataSource, caseInsensitiveOptions, mode)
+      case dataSource: ConsoleSinkProvider =>
+        data.show(data.count().toInt, false)
```

Comment:
`ConsoleSink`  [has two options](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/console.scala#L27-L30) that could be used here -- `numRows` and `truncate`.